### ISSUE TITLE
Object.assign does not throw error when assign value to non-writable property.

### ISSUE THREAD

**Issue by krave1986** [facebook/react-native#22519]
## Environment

  React Native Environment Info:
    System:
      OS: Windows 10
      CPU: (4) x64 Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
      Memory: 779.73 MB / 7.91 GB
    Binaries:
      npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.2.0.0 AI-181.5540.7.32.5056338

## Reproducible Demo
```
var target = Object.defineProperty({}, 'foo', {value: 1, writable: false});
Object.assign(target, {foo: 2});
```

## Description

As illustrated on MDN doc and according to the behavior of latest version of chomr, when trying to assign a property which is non-writable in target from source to target, Object.assign should throw a TypeError like:

> TypeError: Cannot assign to read only property 'foo' of object '#<Object>'

**Comment by lemiesz** [comment]
Hmm so I just tried this out, and although it does not throw an error in the React runtime. 
It fails silently and the value of target.foo = 1. 

What is interesting is that I would expect this to throw an error in the debugger regardless. Since all the debugger code is suppose to run in [chrome](https://facebook.github.io/react-native/docs/javascript-environment)

**Comment by Salakar** [comment]
@krave1986 thanks for raising this issue 🎉 

I can confirm that since the new JSC landed on Android (RN v0.59.x) this now behaves correctly and throws a valid error.
